### PR TITLE
refactor: modernize loops in hdrgen.d

### PR DESCRIPTION
**Pull request by WalterBright**
also use type inference heavily to ease transition to const correctness

### CODE REVIEW

File: src/hdrgen.d [hunk 1]
```diff
@@ -1526,17 +1517,17 @@ public:
...
         else if (Dsymbol s = isDsymbol(oarg))
         {
-            const(char)* p = s.ident ? s.ident.toChars() : s.toChars();
+            const p = s.ident ? s.ident.toChars() : s.toChars();
             buf.writestring(p);
         }
-        else if (Tuple v = isTuple(oarg))
+        else if (auto v = isTuple(oarg))
         {
-            Objects* args = &v.objects;
```
Comment (yebblies) [inline src/hdrgen.d]: I wonder why this was taking the address.

Comment (ibuclaw) [inline src/hdrgen.d]: Eliding a copy, perhaps?

Comment (yebblies) [inline src/hdrgen.d]: mmm

**Comment by yebblies** [comment]
Auto-merge toggled on

**Pull request merged**